src/librustc_typeck/check/regionck.rs

```diff
@@ -1795,7 +1795,7 @@ impl<'a, 'gcx, 'tcx> RegionCtxt<'a, 'gcx, 'tcx> {
                     let (outlives, _) =
                         self.replace_late_bound_regions_with_fresh_var(
                             span,
-                            infer::AssocTypeProjection(projection_ty.item_name(self.tcx)),
+                            infer::AssocTypeProjection(self.tcx.associated_item(projection_ty.item_def_id).name),
```

Comment:
Done. I don't buy the forgetting argument, filing these easy issues should work well for Rust since there are usually enough people looking for an easy first PR. Either way, I intend to get to the two FIXMEs in this review after this merges.